From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Juan Sanchez Fuentes <juanesf91@gmail.com>
Date: Sun, 31 Aug 2026 13:40:00 -0400
Subject: [PATCH] arm64: dts: cubie-a5e: enable the E906 remoteproc

Add the reserved-memory regions for the E906 RISC-V MCU (riscvsram0/1,
DRAM, vring buffers) and enable &e906_rproc with the mailboxes and
memory mappings, mirroring the avaota-a1 board.

Without the riscvsram0/1 reservations the E906 remoteproc driver
cannot probe and the MCU SRAM is left unreserved.

Signed-off-by: Juan Sanchez Fuentes <juanesf91@gmail.com>
Assisted-by: opencode:muse-spark-1.2-contributor-free
---
 arch/arm64/boot/dts/allwinner/sun55i-a527-cubie-a5e.dts | 47 ++++++++++++++
 1 file changed, 47 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun55i-a527-cubie-a5e.dts b/arch/arm64/boot/dts/allwinner/sun55i-a527-cubie-a5e.dts
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/allwinner/sun55i-a527-cubie-a5e.dts
+++ b/arch/arm64/boot/dts/allwinner/sun55i-a527-cubie-a5e.dts
@@ -38,6 +38,37 @@
 		 * just above the kernel image (ends ~0x42c70000) and the RCQ
 		 * pool (~0x43000000), below the dtb (~0x49ff6000).
 		 */
+		riscvsram0_reserved: riscvsram0@7280000 {
+			reg = <0x0 0x07280000 0x0 0x40000>;
+			no-map;
+		};
+
+		riscvsram1_reserved: riscvsram1@72c0000 {
+			reg = <0x0 0x072c0000 0x0 0x40000>;
+			no-map;
+		};
+
+		e906_dram_reserved: e906_dram@60000000 {
+			reg = <0x0 0x60000000 0x0 0xa00000>;
+			no-map;
+		};
+
+		rv_vdev0buffer: vdev0buffer@4ae00000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0x4ae00000 0x0 0x40000>;
+			no-map;
+		};
+
+		rv_vdev0vring0: vdev0vring0@4ae40000 {
+			reg = <0x0 0x4ae40000 0x0 0x2000>;
+			no-map;
+		};
+
+		rv_vdev0vring1: vdev0vring1@4ae42000 {
+			reg = <0x0 0x4ae42000 0x0 0x2000>;
+			no-map;
+		};
+
 		linux,cma {
 			compatible = "shared-dma-pool";
 			reusable;
@@ -608,3 +639,19 @@
 		remote-endpoint = <&hdmi_con_in>;
 	};
 };
+
+&e906_rproc {
+	mboxes = <&msgbox 8>, <&msgbox 10>;
+	mbox-names = "arm-kick", "arm-standby";
+	memory-region = <&riscvsram0_reserved>, <&riscvsram1_reserved>,
+			<&rv_vdev0buffer>, <&rv_vdev0vring0>,
+			<&rv_vdev0vring1>, <&e906_dram_reserved>;
+	memory-mappings =
+		< 0x20000	0x20000		0x20000 >,
+		< 0x40000	0x24000		0x40000 >,
+		< 0x8000000	0x37f00000	0x8000000 >,
+		< 0x3ffc0000	0x40000		0x07280000 >,
+		< 0x40000000	0x40000		0x072c0000 >,
+		< 0x40040000	0x3ffc0000	0x40040000 >;
+	status = "okay";
+};
-- 
Armbian
